### PR TITLE
fix rapidxml to compile with gcc

### PR DESCRIPTION
**Pull request by carmichaelong**
This PR adds forward declarations necessary for GCC 4.7 and above. Also adds a missing #include. @tgeijten

### CODE REVIEW

**Comment by tgeijten** [comment]
Thanks for the update and sorry for the slow response! W.R.T. rapidxml, is it possible to take the forward declarations out of `rapidxml_print.hpp` and move them right before when `rapidxml_print.hpp` is included? I would like to keep 3rd party libs clean in case of updates.

W.R.T. flut: the include you added shouldn't be needed in the current version, can you verify this?

**Comment by carmichaelong** [comment]
I've updated the PR to move the forward declaration out and keep rapidxml clean.

I found that I still needed the include (for `mid_str`), so that is still in this PR. I updated it with some other fixes that are needed for GCC with some of the recent commits.

File: flut/buffer/storage.hpp [hunk 1]
```diff
@@ -23,7 +23,7 @@ namespace flut
 		const L& get_label( index_t channel ) const { return labels_[ channel ]; }
 
 		/// find index of a label
-		index_t find_channel( const L& label ) const { auto it = label_indices_.find( label ); return ( it != label_indices_.end() ) ? it->second : no_index; }
+		index_t find_channel( const L& label ) const { auto it = labels_.find( label ); return ( it != labels_.end() ) ? it->second : no_index; }
```
Comment (tgeijten) [inline flut/buffer/storage.hpp:26]: Just noticed this after merging: it doesn't seem right, could you clarify?

**Review by tgeijten — COMMENTED** [review]
Thanks for the commits, I just have a single question on your change to storage.hpp.

**Comment by tgeijten** [comment]
I fixed this on the master branch, can you check if it still compiles with gcc?

**Comment by carmichaelong** [comment]
This issue looks fixed on gcc, though gcc will not compile right now due to new code that was added. I will put in a new issue about this.

**Pull request merged**